patches/minecraft/net/minecraft/world/DimensionType.java.patch

```diff
@@ -0,0 +1,54 @@
++      this(p_i241973_1_, p_i241973_2_, p_i241973_3_, p_i241973_4_, p_i241973_5_, p_i241973_6_, p_i241973_8_, p_i241973_9_, p_i241973_10_, p_i241973_11_, p_i241973_12_, p_i241973_13_, p_i241973_14_, p_i241973_15_, p_i241973_16_, p_i241973_17_, java.util.Collections.EMPTY_LIST);
++   }
++
++   protected DimensionType(OptionalLong p_i241973_1_, boolean p_i241973_2_, boolean p_i241973_3_, boolean p_i241973_4_, boolean p_i241973_5_, double p_i241973_6_, boolean p_i241973_8_, boolean p_i241973_9_, boolean p_i241973_10_, boolean p_i241973_11_, boolean p_i241973_12_, int p_i241973_13_, IBiomeMagnifier p_i241973_14_, ResourceLocation p_i241973_15_, ResourceLocation p_i241973_16_, float p_i241973_17_, java.util.List<net.minecraftforge.common.world.IDimensionExtra> extras) {
+       this.field_236010_o_ = p_i241973_1_;
+       this.field_218273_h = p_i241973_2_;
+       this.field_236011_q_ = p_i241973_3_;
+@@ -101,6 +110,7 @@
+       this.field_242709_C = p_i241973_16_;
+       this.field_236017_x_ = p_i241973_17_;
+       this.field_236018_y_ = func_236020_a_(p_i241973_17_);
++      this.extras = extras;
+    }
+ 
+    private static float[] func_236020_a_(float p_236020_0_) {
+@@ -272,4 +282,12 @@
+          return this.field_218273_h == p_242714_1_.field_218273_h && this.field_236011_q_ == p_242714_1_.field_236011_q_ && this.field_236012_r_ == p_242714_1_.field_236012_r_ && this.field_236013_s_ == p_242714_1_.field_236013_s_ && this.field_242713_t == p_242714_1_.field_242713_t && this.field_236015_u_ == p_242714_1_.field_236015_u_ && this.field_241499_s_ == p_242714_1_.field_241499_s_ && this.field_241500_t_ == p_242714_1_.field_241500_t_ && this.field_241501_u_ == p_242714_1_.field_241501_u_ && this.field_241502_v_ == p_242714_1_.field_241502_v_ && this.field_241503_w_ == p_242714_1_.field_241503_w_ && Float.compare(p_242714_1_.field_236017_x_, this.field_236017_x_) == 0 && this.field_236010_o_.equals(p_242714_1_.field_236010_o_) && this.field_227175_i_.equals(p_242714_1_.field_227175_i_) && this.field_241504_y_.equals(p_242714_1_.field_241504_y_) && this.field_242709_C.equals(p_242714_1_.field_242709_C);
+       }
+    }
++
++   public java.util.List<net.minecraftforge.common.world.IDimensionExtra> getExtras(){ return extras; }
++   public Optional<net.minecraftforge.common.world.IDimensionExtra> getExtra(ResourceLocation name) {
++      return getExtras().stream().filter(e -> e.getType().getRegistryName().equals(name)).findAny();
++   }
++   public Optional<net.minecraftforge.common.world.IDimensionExtra> getExtra(net.minecraftforge.common.world.DimensionExtraType<?> codec) {
++      return getExtras().stream().filter(e -> e.getType().equals(codec)).findAny();
++   }
```

Comment:
This could do the unchecked cast here to make the return type nicer. Every time a modder uses this they're going to have to cast down from `IDimensionExtra` anyway, and if you only allow querying by type it's even simpler.

```java
public <E extends net.minecraftforge.common.world.IDimensionExtra> Optional<E> getExtra(net.minecraftforge.common.world.DimensionExtraType<?> type) {
    return getExtras().stream().filter(e -> e.getType().equals(type)).findAny().map(e -> (E) e);
}
```

Reply:
Actually if you do this it might as well be a `Map<DimensionExtraType, IDimensionExtra>` if you're already requiring them to be unique per type.